libraries/bsg_manycore_epa.h

```diff
@@ -51,6 +54,33 @@ extern "C" {
 #define EPA_FROM_BASE_AND_OFFSET(base, offset)  \
         (((base)+(offset)))
 
+        /**
+         * Checks alignment of an epa based on data size in bytes.
+         * @param[in] epa  epa address
+         * @param[in] sz   data size in bytes.
+         * @return         HB_MC_SUCCESS if npa is aligned and HB_MC_INVALID otherwise.
+         */
+        static inline int hb_mc_epa_check_alignment(const hb_mc_epa_t *epa, size_t sz)
+        {
+                switch (sz) {
+                case 4:
```

Comment:
Macro values would be used here too

Reply:
I'm not sure I agree that a macro value should be used here. @drichmond why do you think a macro should be used?

Reply:
Here, specifically? Or in general on the 4/2/1 values? 

My opinion is that we're using 4/2/1 repeatedly in this code. We use it in the writeN functions, we use it in the alignment checker, and we use it in the write() function. It feels gratuitous to me, hence the macro

Reply:
Are you suggesting this code become
```
case FOUR:
 // ...
case TWO:
// ...
case ONE:
// ...
```
Or am I misunderstanding you?

Reply:
I agree with you, I was just making a (bad) joke

Reply:
OK, I wasn't sure if this conversation was also resolved.